src/test/java/io/vertx/core/shareddata/AsyncMapTest.java

```diff
@@ -457,6 +457,19 @@ public void testMapReplaceClusterSerializableObject() {
     testMapReplace(new SomeClusterSerializableObject("foo"), new SomeClusterSerializableObject("bar"), new SomeClusterSerializableObject("quux"));
   }
 
+  @Test
+  public void testMapReplaceTtl() {
+    getVertx().sharedData().<String, String>getAsyncMap("foo", onSuccess(map -> {
```

Comment:
Can you also test the other case (`replace` returns a `Future` completed with previous value)?